### ISSUE TITLE
🐛Error message "Something went wrong" during checkout

### ISSUE THREAD

**Issue by cobalt-tendons** [HiEventsDev/Hi.Events#253]
**Describe the bug**
I have deployed on Render and everything works fine using the default project URL they assign automatically, e.g. hi-events-xxxx.onrender.com.  I am able to manage events and checkout without any issues.  However, I cannot complete the checkout when using a custom domain, e.g. rsvp.mydomain.com.

**To Reproduce**
1. Go to Hi-Events instance at rsvp.mydomain.com
2. Add tickets to cart
3. Click Continue button
4. See error (screenshot attached)

**Expected behavior**
Expected to continue to the next step of the checkout process

**Screenshots**
![Screenshot 2024-10-14 3 14 15 PM](https://github.com/user-attachments/assets/eabd9613-de16-4934-87be-40eea57fca05)

**Desktop (please complete the following information):**
 - OS: Chrome OS
 - Browser: Chrome
 - Version: 126.0.6478.252

**Smartphone (please complete the following information):**
 - Device: iPhone12
 - OS: iOS 18
 - Browser: Stock Safari

**Hi.Events Version and platform**
Docker @ render.com

**Logs**
N/A

**Additional context**
N/A

**Comment by daveearley** [comment]
Hi @cobalt-tendons,

From the screenshot it appears the API URL being used is still the default render URL. This may explain the error. In Render.com, what is the value for the `VITE_API_URL_CLIENT` env variable? It should be the custom domain, not the Render default domain.

**Comment by cobalt-tendons** [comment]
Thanks @daveearley and sorry for the delay.  The Environment page in Render didn't have a `VITE_API_URL_CLIENT` variable, so I added it =rsvp.mydomain.com, but it didn't help.  Screenshots attached.  I also hopped into the container, manually exported the variable and restarted the node app, still getting that 403.

![env](https://github.com/user-attachments/assets/dd2172d6-c48a-48f4-9e66-a4aab21bf1ac)
![devtools](https://github.com/user-attachments/assets/21f37b00-0286-488a-a276-b7a212a8229f)

**Comment by daveearley** [comment]
It looks like the API URL still isn't correct. I'm not overly familiar with Render.com, so you may need to reach out to their support to see why the env variable isn't updating.

**Comment by f1dz** [comment]
It has also happened to me. The problem was we were using the same browser to log in as admin and to order the ticket.
Try a different browser or opening a new window on incognito/private solved the problem.

**Comment by daveearley** [comment]
@f1dz Thanks for the additional information. Admins should be able to complete orders without issue. Do you see any errors in the logs when you encounter this error?

**Comment by f1dz** [comment]
> @f1dz Thanks for the additional information. Admins should be able to complete orders without issue. Do you see any errors in the logs when you encounter this error?

Yes, I got the same error when I use the same browser as admin logged in. After using different browser, it went well. I use Safari for Admin and Chrome for attendee.

Additional information, I use docker all-in-one.

**Comment by Caseit18** [comment]
Also getting the 403 "Sorry, we could not verify your session. Please restart your order." (in the console)
The Header shows: GET http://10.139.50.19:8123/api/public/events/1/order/o_AjFOlsECqMNXz?include=

Self Hosted, All In One docker (to get a better understanding of the product). 
I did try separate browsers than the admin one per @f1dz 's suggestion. Problem remains.

(my .env file)
VITE_FRONTEND_URL=http://10.139.50.19:8123
VITE_API_URL_CLIENT=http://10.139.50.19:8123/api
VITE_API_URL_SERVER=http://10.139.50.19:80/api
VITE_STRIPE_PUBLISHABLE_KEY=pk_test
LOG_CHANNEL=stderr
QUEUE_CONNECTION=sync
MAIL_MAILER=log
APP_DISABLE_REGISTRATION=true
APP_SAAS_MODE_ENABLED=false
APP_CDN_URL=http://10.139.50.19:8123/storage

Forgive the very noobish question, but shouldn't there be something after the = in my GET command?

**Comment by daveearley** [comment]
Hi @Caseit18, the issue may be that you're using HTTP, rather than HTTPS. The checkout process relies on a secure cookie to track the customer's session. Some browsers will not allow a secure cookie to be set over HTTP.

**Comment by Caseit18** [comment]
> Hi @Caseit18, the issue may be that you're using HTTP, rather than HTTPS. The checkout process relies on a secure cookie to track the customer's session. Some browsers will not allow a secure cookie to be set over HTTP.

I appreciate the feedback. I'm worried I'm taking this ticket off track.

I punched https in place of the http variables but it now generates an "ERR_SSL_PROTOCOL_ERROR" when trying to access.

VITE_FRONTEND_URL=https://10.139.50.19:8123
VITE_API_URL_CLIENT=https://10.139.50.19:8123/api
VITE_API_URL_SERVER=https://10.139.50.19:80/api 
This feels funky given that the nginx inside of the container is doing 'normal' listening on port 80 (which is 'mapped' via docker from 8123). Is a cert even generated if you don't do the 'local development' path of deployment?

**Comment by ravibhoraniya** [comment]
Hi, I tried without Docker and got the same error. Please check my environment variable.

Front Build
VITE_API_URL_CLIENT=http://localhost:8000
VITE_API_URL_SERVER=http://localhost:8000

VITE_FRONTEND_URL=http://localhost:5678


Backend
APP_URL=http://localhost:5678

**Comment by daveearley** [comment]
@ravibhoraniya With the limited information you've provided it's difficult to determine the issue. Have you followed the instructions in [this guide](https://github.com/HiEventsDev/Hi.Events/blob/develop/INSTALL_WITHOUT_DOCKER.md)?

**Comment by ravibhoraniya** [comment]
Hello @daveearley, I would like to inform you that I have followed the same steps and configured the .env file accordingly; however, I am still encountering this issue. I have attempted this process four times.

here data: 
![Screenshot 2025-01-02 at 4 44 11 PM](https://github.com/user-attachments/assets/7f43feaf-e8a4-4133-b79d-72fe73e899f7)

  VITE_API_URL_CLIENT: 'https://event.iwebinfotech.com',
  VITE_API_URL_SERVER: 'https://event.iwebinfotech.com',
  VITE_FRONTEND_URL: 'http://localhost:3002', 
  NODE_PORT: '3002'

**Comment by xmajzel** [comment]
I did come up to the same exact problem state, everything is set up correctly, but cookie, which should be sent from Backend, when creating order -> returned status of order is  **RESERVED** with **Cookie** in response. But Cookie in response is invalid and browser does not set it up.

Cookie attribute SameSite is lowercase, but should be **None** - uppercase first letter

![Image](https://github.com/user-attachments/assets/0437d93c-8321-490b-87a0-3e2871ae9e81)

**Comment by xmajzel** [comment]
@daveearley

**Comment by daveearley** [comment]
@xmajzel I'm not convinced the case of `None` is the issue here, as from what I can see the backend sends the correct case. I suspect the issue may be CORS exposed headers setting not allowing the `Set-Cookie` header, as seen [here](https://github.com/HiEventsDev/Hi.Events/pull/282/files#diff-fd802201e136915fb7703a159536371a8d5a0b1d519f8592a524ad5742befc31).

There is a fix in the V1.0.0 branch, but I will also pull it into the v8.X branch.

Thanks!

**Comment by xmajzel** [comment]
Oh, you are right! Thank you for response, looking forward to see fix there.

**Comment by xmajzel** [comment]
Okay, I've tested it, but probably problem is completely elsewhere. When looking into frontend code, when creating client in [client.ts](https://github.com/HiEventsDev/Hi.Events/blob/ad049649bf17a7945f1347fd5753fc2481d663a5/frontend/src/api/client.ts#L4), it uses withCredentials:

```typescript
export const api = axios.create({
    baseURL: BASE_URL,
    headers: {
        'Content-Type': 'application/json'
    },
    withCredentials: true,
});
```

But, when creating public client in [public-client.ts](https://github.com/HiEventsDev/Hi.Events/blob/ad049649bf17a7945f1347fd5753fc2481d663a5/frontend/src/api/public-client.ts#L4) it does not:

```typescript
export const publicApi = axios.create();
```

And indeed i think thats causing the problem @daveearley

**Comment by kleyton67** [comment]
Sorry I have duplicated this issue #380 , something has already fixed it?

**Comment by AdoCaice** [comment]
> Sorry I have duplicated this issue [#380](https://github.com/HiEventsDev/Hi.Events/issues/380) , something has already fixed it?
@kleyton67 This is what worked for me. 
Example: 
```

server: {
    proxy: {
      // Only proxy the checkout-related endpoints
      '/public/events/*/order': {
        target: 'http://localhost:8000',
        changeOrigin: true,
      },
      '/public/orders': {
        target: 'http://localhost:8000',
        changeOrigin: true,
      }
    }
  }
```

**Comment by daveearley** [comment]
Hello everyone,

This thread touches on multiple, possibly different issues, all of which appear to have been addressed in the v1.X.X release.

If you're still experiencing any problems, please open a new issue with detailed information about what you're encountering.

Thank you!
